packages/next/src/build/webpack-config.ts

```diff
@@ -473,6 +473,47 @@ export default async function getBaseWebpackConfig(
       )
     }
 
+    const useBuiltinSwcLoader = process.env.BUILTIN_SWC_LOADER
+    if (isRspack && useBuiltinSwcLoader) {
+      return {
+        loader: 'builtin:next-swc-loader',
+        options: {
+          isServer: isNodeOrEdgeCompilation,
+          rootDir: dir,
+          pagesDir,
+          appDir,
+          hasReactRefresh: dev && isClient,
+          // nextConfig: config,
+          // jsConfig,
+          transpilePackages: finalTranspilePackages,
+          supportedBrowsers,
+          swcCacheDir: path.join(
+            dir,
+            config?.distDir ?? '.next',
+            'cache',
+            'swc'
+          ),
+          serverReferenceHashSalt: encryptionKey,
+
+          // rspack specific options
+          pnp: Boolean(process.env.pnp),
```

Comment:
Looks like this is introducing a new environment variable, let's prefix it with `NEXT_`.

```suggestion
          pnp: Boolean(process.env.NEXT_RSPACK_PNP),
```

Reply:
I assumed this was set by yarn when running with pnp -- is this not the case @hardfist?

Reply:
Yea this is the `pnp` variable which we check in our `swc` loader as well although seems this is checking `env` instead of `versions` https://github.com/vercel/next.js/blob/a7d481850365df47a876b6359947338a8cdb662d/packages/next/src/build/webpack/loaders/next-swc-loader.ts#L219